crates/redis_interface/src/lib.rs

```diff
@@ -21,24 +21,20 @@ pub mod commands;
 pub mod errors;
 pub mod types;
 
+use fred::prelude::RedisError;
 use router_env::logger;
 
 pub use self::{commands::*, types::*};
 
 pub struct RedisConnectionPool {
     pub pool: fred::pool::RedisPool,
     config: RedisConfig,
-    _join_handles: Vec<fred::types::ConnectHandle>,
+    join_handles: Vec<fred::types::ConnectHandle>,
 }
 
 impl RedisConnectionPool {
     /// Create a new Redis connection
-    ///
-    /// # Panics
-    ///
-    /// Panics if a connection to Redis is not successful.
-    #[allow(clippy::expect_used)]
-    pub async fn new(conf: &RedisSettings) -> Self {
+    pub async fn new(conf: &RedisSettings) -> Result<Self, RedisError> {
```

Comment:
Do you think we should raise our error type instead, an `errors::RedisError::RedisConnectionError` variant maybe?

Reply:
i guess we could something like 
```rust
RedisConnectionError {err: fred::RedisError}
```

Reply:
Either that, or return an `error_stack::Result` / `CustomResult`.